### DISCUSSION TITLE
Is there any way to update the particles dynamically?

### DISCUSSION THREAD

**Discussion by kabirverman** [tsparticles/react#54] [Q&A] [answered]
Let's say I want to update state and change the particle color. Is there a way I can keep the same particles that are alive on screen and just update their colors? To clarify, I _don't_ want to destroy them and spawn new particles with the correct color.

Any Help would be appreciated!

**Comment by matteobruni** [comment]
Yes, but you need to write a custom plugin for that

**Reply by kabirverman** [reply]
I can't seem to find much documentation other than https://particles.js.org/docs/interfaces/tsParticles_Engine.Core_Interfaces_IContainerPlugin.IContainerPlugin.html

Can you point me to where I can learn more about creating a custom plugin?

**Reply by matteobruni** [reply, marked as answer]
https://github.com/tsparticles/tsparticles/tree/main/updaters/color

This is the plugin responsible for the color updates, it can be a starting point.

**Reply by kabirverman** [reply]
thanks!
